src/components/Dropdown.tsx

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { sortingTypes } from "../App";
+import "../components/DropDown.css"
+import { CurrentSelection } from "./SortControl";
+
+interface IDropDown {
+    setCurrentSelection: (value: (((prevState: CurrentSelection) => CurrentSelection) | CurrentSelection)) => void
```

Comment:
I suggest simplify this type and use just `(newValue: CurrentSelection ) => void` or React's type for setters.

Reply:
fixed